lib/literal/array.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+class Literal::ArrayGeneric
+	def initialize(type)
+		@type = type
+	end
+
+	def new(*value)
+		Literal::Array.new(@type, value)
+	end
+
+	alias_method :[], :new
+
+	def ===(value)
+		Literal::Array === value && @type == value.__type__
+	end
+end
+
+class Literal::Array
+	def initialize(type, value)
+		unless Array === value && value.all?(type)
+			raise
+		end
+
+		@__type__ = type
+		@__value__ = value
+	end
```

Comment:
We should probably allocate and call another method because this shouldn’t be called directly by a user.